src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -140,25 +141,24 @@ internal static bool DoSslHandshake(IntPtr sslContextPtr, IntPtr recvPtr, int re
                     error = sendCount;
                     Marshal.FreeHGlobal(sendPtr);
                     sendPtr = IntPtr.Zero;
-                    sendCount = 0;
-                    error = GetSslError(context.sslPtr, error, "Read Bio failed: ");                 
-                    throw CreateSslException("SSL Handshake failed", error);                   
+                    sendCount = 0;                  
+                    throw CreateSslException(context.sslPtr, "Read Bio failed: ", error);                   
                 }
             }
         
-            return ((libssl.SSL_state(context.sslPtr) == libssl.SslState.SSL_ST_OK)) ? true : false;
+            return ((libssl.SSL_state(context.sslPtr) == (int)libssl.SslState.SSL_ST_OK));
```

Comment:
The extern method can be declared to return the SslState enum just fine (the marshaller effectively just returns the result of casting the int).  Just like casting an int to an enum normally, unmapped values will succeed the cast, they're just hard to talk about and won't pretty-print.

That can always be done in a later change, though.